### PR TITLE
Pom refactor

### PR DESCRIPTION
**Pull request by magdalsen**

### CODE REVIEW

File: playwright.config.ts [hunk 1]
```diff
@@ -1,4 +1,8 @@
-import { defineConfig, devices } from '@playwright/test';
+import { defineConfig, devices } from "@playwright/test";
+import dotenv from "dotenv";
+dotenv.config({
+  path: ".env"
```
Comment (Vojow) [inline playwright.config.ts]: Dotenv by default looks for this path

File: fixtures/calendar.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { test } from "@playwright/test";
+
+type Calendar = {
```
Comment (Vojow) [inline fixtures/calendar.ts]: instead of type you can use interface

File: myTests/calendar.spec.ts [hunk 3]
```diff
@@ -1,43 +1,40 @@
...
-    "div[class='datepicker-days'] th[class='datepicker-switch']"
-  );
-
-  let dateToSelect: string = "December 2022";
+calendarTest("Check if calendar data is fullfilled", async ({ page, date }) => {
+  const navigatePage = new NavigatePage(page);
+  const calendarPage = new CalendarPage(page);
 
-  const thisMonth = moment(dateToSelect, "MMMM YYYY").isBefore();
+  await navigatePage.navigateToURL(`${process.env.CALENDAR_URL}`);
```
Comment (Vojow) [inline myTests/calendar.spec.ts]: by default values returned from this env are strings

File: shared/navigate-page.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { Page } from "@playwright/test";
+import { BasePage } from "./base-page";
+
+export class NavigatePage extends BasePage {
+  async navigateToBaseURL() {
```
Comment (Vojow) [inline shared/navigate-page.ts:4]: you could create single method, but it's good for now

File: pages/dropdown-page.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import { Page } from "@playwright/test";
+import { BasePage } from "../shared/base-page";
+
+export class DropdownPage extends BasePage {
+  get multiSelectList() {
+    return "#multi-select";
+  }
+
+  get printAllButton() {
+    return 'div.input-body button[id="printAll"]';
```
Comment (Vojow) [inline pages/dropdown-page.ts]: for id there is different way to write it, which you already used

File: pages/calendar-page.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  get calendarData() {
+    return "input[id='birthday']";
+  }
+
+  get startDate() {
+    return "input[placeholder='Start date']";
+  }
+
+  get previousDate() {
+    return "div[class='datepicker-days'] th[class='prev']";
```
Comment (Vojow) [inline pages/calendar-page.ts]: for class there is different way to write it, which you already used

File: myTests/send-email.spec.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+      await sendEmailPage.fillInputFieldEmail(email);
+      await sendEmailPage.fillInputFieldSubject(subject);
+      await sendEmailPage.fillInputFieldMessage(message);
+      const submitButton = page.locator(sendEmailPage.submitButton);
+      await submitButton.scrollIntoViewIfNeeded();
+      page.on("dialog", async (dialog) => {
+        await dialog.accept();
+      });
+      await sendEmailPage.clickSubmitButton();
+      const successMessage = page.locator(sendEmailPage.successMessage);
```
Comment (Vojow) [inline myTests/send-email.spec.ts]: IF you would operate on WebElements, not selectors, you wouldn't need to use page.locator here

File: myTests/checkbox.spec.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+  let navigatePage: NavigatePage;
+
+  test.beforeEach(async ({ browser }) => {
+    page = await browser.newPage();
+    navigatePage = new NavigatePage(page);
+    await navigatePage.navigateToLambdatestPage();
+  });
+
+  test("Checkbox test", async () => {
+    const singleCheckbox = page.locator("id=isAgeSelected");
```
Comment (Vojow) [inline myTests/checkbox.spec.ts:15]: No Page Object Pattern here

File: myTests/CRUD-pet-store.spec.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
+import test, { expect } from "@playwright/test";
+import { petShopResponses } from "../data/pet-shop-responses";
+const ID_find = 10;
+const ID_delete = 12;
+
+test.describe.parallel("API testing", () => {
+  test("Add a new pet to the store", async ({ request }) => {
+    const newPet = await request.post(`${process.env.BASE_URL_PETSTORE}/pet`, {
```
Comment (Vojow) [inline myTests/CRUD-pet-store.spec.js]: so this url duplicated multiple times

File: myTests/CRUD-pet-store.spec.js [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+      data: petShopResponses.post.newPetDataRequest
+    });
+    expect(newPet.ok()).toBeTruthy();
```
Comment (Vojow) [inline myTests/CRUD-pet-store.spec.js]: why do you need this assertion?

**Review by Vojow — CHANGES_REQUESTED** [review]

File: pages/calendar-page.ts (hunk 6 above)
Comment (magdalsen) [reply pages/calendar-page.ts]: I'm not sure what you have on mind. Something like this?
  get previousDate() {
    return "div.datepicker-days th.prev";
  }

File: pages/dropdown-page.ts (hunk 5 above)
Comment (magdalsen) [reply pages/dropdown-page.ts]: Do you mean:
return 'div.input-body button#printAll' ?

**Review by Vojow — APPROVED** [review]

**Comment by Vojow** [comment]
Create another branch with changes to POM and API testing

**Pull request merged**